_posts/2022-04-21-nextjs_frontend_framework_with_be_api.md

```diff
@@ -0,0 +1,151 @@
+---
+layout: post
+title: How to setup NextJS FrontEnd Framework with some API? 
+categories: [NextJS]
+author_github: nevermore17
+author: Verbin Kirill
+---
+
+Let's say we have some BackEnd server (in my case, Django + Django Rest Framework) and want to generate `html`/`css` on NextJS
+
+## 1. Create NextJS app
+ ``` 
+    npx create-next-app myapp
+ ```
+NextJS application using a folder structure such as:
+```
+myapp
+|-- pages
+|   |-- api
+|   |   |-- hello.js
+|   |
+|   |-- _app.js
+|   |-- index.js
+|
+|-- public
+|   |-- favicon.ico
+|  
+|-- styles
+|   |-- global.css
+|   |-- Home.module.css
+|
+|--package.json
+
+```
+On this step I recommend add to myapp folder files `.env.development` and `.env.production`. Depending on the start mode(dev or prod) NextJs add values from `.end.<start_mode>` files to `process.env`. This will make development and deployment easier.
```

Comment:
In this step I recommend to add to myapp folder two files named `.env.development` and `.env.production`. Depending on the start mode (dev or prod), NextJs adds values from `.end.<start_mode>` files to `process.env`. This will make development and deployment easier.